src/Native/Unix/System.Native/pal_signal.c

```diff
@@ -108,6 +111,15 @@ static void* SignalHandlerLoop(void* arg)
             return NULL;
         }
 
+        if (signalCode == SIGCHLD || signalCode == SIGCONT || signalCode == SIGWINCH)
```

Comment:
Are there other things that could invalidate the cursor position?  For example:
- Are there any terminals in which clicking the mouse at an arbitrary location in the terminal would effectively move the cursor to that location?
- What about commands that move the cursor, e.g. if someone wrote out the ANSI sequence for setting the cursor position?
- What if someone clears the terminal using a command on their terminal window?

Etc.

Reply:
> Are there any terminals in which clicking the mouse at an arbitrary location in the terminal would effectively move the cursor to that location?

I'd expect the application would anyhow need to indicate to the terminal it supports that.

> What if someone clears the terminal using a command on their terminal window?

It's not possible to detect this. Applications that position their text don't like it when that happens.

Reply:
> It's not possible to detect this. Applications that position their text don't like it when that happens.

Do you know what curses does in this case?  Presumably it caches as well.  Do apps that use it just end up being garbled until something causes then to re-query for the actual position from the terminal?

I just want to make sure we're using all available signals to aggressively invalidate.

Reply:
> Do you know what curses does in this case? Presumably it caches as well. Do apps that use it just end up being garbled until something causes then to re-query for the actual position from the terminal?

I haven't checked the ncurses code. There is no signal for it. `nano`, `vi` and `mc` don't deal with it and end up being garbled. Resizing the window (SIGWINCH) gets them to a proper state.

Reply:
> nano, vi and mc don't deal with it and end up being garbled

Hmm.  Ok.  I guess if prominent apps like that have the same issue, it's at least semi-explainable as a general problem across anything on the platform.

Reply:
> There is no signal for it.

FWIW, I meant "signal" in the general sense, i.e. something that would provide a notification or that we could cheaply query or some such thing, not just actual unix signals.

Reply:
I think we are using everything we can. We should be as good as other applications.

We may want to expose WIGWINCH at some point. I once created [an issue for that](https://github.com/dotnet/corefx/issues/14152), but we closed it due to a lack of interest.